ebpf/src/ebpfprobe.cpp

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2019-present Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ebpfprobe.h"
+
+#include <osquery/logger.h>
+
+namespace trailofbits {
+namespace {
+const int kPollTime{1000};
+
+using BPFRef = std::unique_ptr<ebpf::BPF>;
+
+void getProbeEventName(std::string& event_name,
+                       const eBPFProbeDescriptor::Probe& probe) {
+  event_name = {};
```

Comment:
Is there a performance/behavior difference between this and `clear`?

Reply:
Godbolt suggests that `operator=` with `{}` requires some additional allocations: https://godbolt.org/z/nIv-2Q

Obviously still a minor nit 😄